tests/integration/conftest.py

```diff
@@ -645,24 +527,24 @@ def environment_variables(request):
     """
     Fixture.
     Sets and unsets required environment variables.
-    Environment variable(s) needs to be passed as a list(s) to pytest parametrize.
     Usage:
-    @pytest.mark.parametrize("envars", [["LIST", "OF"], ["ENVIRONMENT", "VARIABLES"]])
+    @pytest.mark.parametrize("environment_variables", ["CONVERT2RHEL_UNSUPPORTED], indirect=True)
+    @pytest.mark.parametrize("environment_variables", [["LIST", "OF"], ["ENVIRONMENT", "VARIABLES"]], indirect=True)
     """
 
-    def _set_env_var(envars):
-        for envar in envars:
-            os.environ[envar] = "1"
+    if hasattr(request, "param"):
+        env_vars = request.param
+        if not isinstance(env_vars, list):
+            env_vars = [env_vars]
 
-    yield _set_env_var
+        for e in env_vars:
+            os.environ[e] = "1"
```

Comment:
If the request.param can contain nested lists as added in the function description then this for will fail

Reply:
Ok then there is no problem with the code. Thanks for the explanation and updating the function comment.